### PR TITLE
G Suite: Prevent users from adding G Suite to mapped domains without WordPress.com nameservers

### PR DESCRIPTION
**Pull request by olaseni**
#### Changes proposed in this Pull Request

Introduces:
- `isMappedDomainWithWpcomNameservers` which is an update to `isMappedDomain` that filters for mapped domains that are configured with WPcom nameservers

#### Testing instructions

_This PR **works** in collaboration with D33844-code._

- Attempt to replicate issue #36211 
- Confirm that you cannot purchase G Suite when domain is mapped and has not been configured with WordPress.com nameservers.
- You should see screen instead:

Screenshot 2019-10-24 at 5 07 10 AM

### CODE REVIEW

File: client/lib/gsuite/index.js [hunk 1]
```diff
@@ -107,10 +107,11 @@ function getGSuiteSupportedDomains( domains ) {
...
 		}
 
-		const isHostedOnWpcom = isRegisteredDomain( domain ) && ( domain.hasWpcomNameservers || hasGSuiteWithUs( domain ) );
-		const isMapped = isMappedDomain( domain );
+		const isHostedOnWpcom =
+			isRegisteredDomain( domain ) && ( domain.hasWpcomNameservers || hasGSuiteWithUs( domain ) );
+		const isMappedWithWpcomNameservers = isMappedDomainWithWpcomNameservers( domain );
 
-		return ( isHostedOnWpcom || isMapped ) && canDomainAddGSuite( domain.name );
+		return ( isHostedOnWpcom || isMappedWithWpcomNameservers ) && canDomainAddGSuite( domain.name );
```
Comment (stephanethomas) [inline client/lib/gsuite/index.js]: We could avoid the unnecessary `isMappedWithWpcomNameservers` variable by structure our code differently:

```javascript
if ( ! isHostedOnWpcom && ! isMappedDomainWithWpcomNameservers( domain ) ) {
  return false;
}

return canDomainAddGSuite( domain.name );
```

File: client/my-sites/email/email-management/index.jsx [hunk 2]
```diff
@@ -137,6 +137,17 @@ class EmailManagement extends React.Component {
 				secondaryAction: translate( 'Add Email Forwarding' ),
 				secondaryActionURL: emailManagementForwarding( selectedSiteSlug, selectedDomainName ),
 			};
+			if ( isMappedDomain( selectedDomain ) ) {
+				Object.assign( emptyContentProps, {
+					line: translate(
+						'Only domains using WordPress.com name servers are eligible for G Suite.'
+					),
+					action: translate( 'How To Change Name Servers' ),
```
Comment (a8ci18n) [inline client/my-sites/email/email-management/index.jsx]: Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'How to change your name servers' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/email/email-management/index.jsx [hunk 3]
```diff
@@ -137,6 +137,19 @@ class EmailManagement extends React.Component {
 				secondaryAction: translate( 'Add Email Forwarding' ),
 				secondaryActionURL: emailManagementForwarding( selectedSiteSlug, selectedDomainName ),
 			};
+			if ( isMappedDomain( selectedDomain ) ) {
+				Object.assign( emptyContentProps, {
+					title: translate( 'G Suite is not supported on this domain yet.' ),
```
Comment (a8ci18n) [inline client/my-sites/email/email-management/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'G Suite is not supported on this domain' )` *ES Score: 7*

File: client/my-sites/email/email-management/index.jsx [hunk 4]
```diff
@@ -137,6 +137,19 @@ class EmailManagement extends React.Component {
 				secondaryAction: translate( 'Add Email Forwarding' ),
 				secondaryActionURL: emailManagementForwarding( selectedSiteSlug, selectedDomainName ),
 			};
+			if ( isMappedDomain( selectedDomain ) ) {
+				Object.assign( emptyContentProps, {
+					title: translate( 'G Suite is not supported on this domain' ),
```
Comment (stephanethomas) [inline client/my-sites/email/email-management/index.jsx]: Why do we override this `title` property with the same value already present in `emptyContentProps`?

File: client/my-sites/email/email-management/index.jsx [hunk 5]
```diff
@@ -137,6 +137,19 @@ class EmailManagement extends React.Component {
...
+					line: translate(
+						'To enable G Suite on %(domain)s, please have its name servers changed to that of WordPress.com.',
+						{ args: { domain: selectedDomainName } }
+					),
+					action: translate( 'How to change your name servers' ),
+					actionURL:
+						'https://en.support.wordpress.com/domains/map-existing-domain/#change-your-domains-name-servers',
```
Comment (stephanethomas) [inline client/my-sites/email/email-management/index.jsx]: Let's avoid the `en.` subdomain to better serve non-English users (or maybe use the [`localizeUrl()` function](https://github.com/Automattic/wp-calypso/blob/bd07030f74ff1cfdc9e5929d23811d18bef0c4c0/client/lib/i18n-utils/utils.js#L180)).

Comment (olaseni) [reply client/my-sites/email/email-management/index.jsx]: Thanks -  I missed that

**Comment by stephanethomas** [comment]
I've addressed my comments, and simplified the code a bit.

**Comment by stephanethomas** [comment]
I've rebased this pull request against master.

**Pull request merged**